### PR TITLE
Bump dex to 0.4.0 and remove pkger

### LINKED ISSUE #6563: Switch Dex fork to Go 1.16 embedding, eliminate pkger
* Update `http.FileSystem` PR to upstream Dex: https://github.com/dexidp/dex/pull/1416
* Cut a new tag for `concourse/dex` and bump dependency in `go.mod`

### PR DESCRIPTION
**Pull request by aoldershaw**
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## What does this PR accomplish?

Bug Fix | Feature | Documentation

closes #6563 .

## Changes proposed by this PR:

* Bump dex to 0.4.0, which includes the new dexidp/dex release 2.28.0
  * We have a PR for including an http filesystem which previously used pkger to bundle the assets, but now uses Go 1.16's file embedding
  * There were also some changes to the template - now, assets are referenced using a relative path, so we don't need to specify the paths relative to `/sky/issuer`

## Notes to reviewer:

`go mod tidy` picked up a bunch of other dependency updates as well, so we just left them in

## Contributor Checklist

- [ ] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [ ] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -65,18 +59,17 @@ require (
 	github.com/nu7hatch/gouuid v0.0.0-20131221200532-179d4d0c4d8d
 	github.com/onsi/ginkgo v1.14.2
 	github.com/onsi/gomega v1.10.3
-	github.com/opencontainers/runtime-spec v1.0.1
+	github.com/opencontainers/runtime-spec v1.0.3-0.20200929063507-e6143ca7d51d
```
[the remark this answers is not in the document]
Comment (chenbh) [reply go.mod:62]: Actually it looks like gomod resorts to this format when a repo doesn't follow the gomodules convention, but it's weird how it picked up `v1.0.3` (all the other ones are `v0.0.0-...`)

I would rollback to `v1.0.2` just for our sanity.

**Review by chenbh — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -18,27 +19,21 @@ require (
...
 	github.com/concourse/go-archive v1.0.1
 	github.com/concourse/retryhttp v1.1.0
-	github.com/containerd/cgroups v0.0.0-20191220161829-06e718085901 // indirect
-	github.com/containerd/containerd v1.3.4
-	github.com/containerd/continuity v0.0.0-20191214063359-1097c8bae83b // indirect
-	github.com/containerd/fifo v0.0.0-20191213151349-ff969a566b00 // indirect
-	github.com/containerd/go-cni v0.0.0-20200107172653-c154a49e2c75
-	github.com/containerd/ttrpc v0.0.0-20191028202541-4f1b8fe65a5c // indirect
-	github.com/containerd/typeurl v0.0.0-20190911142611-5eb25027c9fd
+	github.com/containerd/containerd v1.5.0-beta.1
```
[the remark this answers is not in the document]
Comment (xtremerui) [reply go.mod:26]: it passes!

**Comment by xtremerui** [comment]
@chenbh both are reverted

**Review by chenbh — APPROVED** [review]
That last commit is still authored and signed off by `Frances Bar <bar@awesometown.local>`, but otherwise lgtm

**Pull request merged**